pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/upsertcompaction/UpsertCompactionTaskGenerator.java

```diff
@@ -155,6 +159,10 @@ public List<PinotTaskConfig> generateTasks(List<TableConfig> tableConfigs) {
 
       SegmentSelectionResult segmentSelectionResult =
           processValidDocIdsMetadata(taskConfigs, completedSegmentsMap, validDocIdsMetadataList);
+      LOGGER.info("Selected {} segments for compaction, {} segments for deletion and skipped {} segments for table: {}",
+          segmentSelectionResult.getSegmentsForCompaction().size(),
+              segmentSelectionResult.getSegmentsForDeletion().size(), segmentSelectionResult._skippedSegments.size(),
+              tableNameWithType);
```

Comment:
Instead of adding skippedSegments var for just this log line, we can calculate skippedSegments size here itself using `validDocIdsMetadataList.size() - segmentSelectionResult.getSegmentsForCompaction().size() - segmentSelectionResult.getSegmentsForDeletion().size()`